overlord/managers_test.go

```diff
@@ -130,71 +131,77 @@ func verifyLastTasksetIsRerefresh(c *C, tts []*state.TaskSet) {
 	c.Check(ts.Tasks()[0].Kind(), Equals, "check-rerefresh")
 }
 
-func (ms *mgrsSuite) SetUpTest(c *C) {
-	ms.tempdir = c.MkDir()
-	dirs.SetRootDir(ms.tempdir)
+func (s *mgrsSuite) SetUpTest(c *C) {
+	s.BaseTest.SetUpTest(c)
+
+	s.tempdir = c.MkDir()
+	dirs.SetRootDir(s.tempdir)
 	err := os.MkdirAll(filepath.Dir(dirs.SnapStateFile), 0755)
 	c.Assert(err, IsNil)
 
 	// needed by hooks
-	ms.mockSnapCmd = testutil.MockCommand(c, "snap", "")
+	s.AddCleanup(testutil.MockCommand(c, "snap", "").Restore)
 
 	oldSetupInstallHook := snapstate.SetupInstallHook
 	oldSetupRemoveHook := snapstate.SetupRemoveHook
 	snapstate.SetupRemoveHook = hookstate.SetupRemoveHook
 	snapstate.SetupInstallHook = hookstate.SetupInstallHook
+	s.AddCleanup(func() {
+		snapstate.SetupRemoveHook = oldSetupRemoveHook
+		snapstate.SetupInstallHook = oldSetupInstallHook
+	})
 
-	ms.automaticSnapshots = nil
-	restoreBackendSave := snapshotstate.MockBackendSave(func(_ context.Context, id uint64, si *snap.Info, cfg map[string]interface{}, usernames []string, flags *snapshotbackend.Flags) (*client.Snapshot, error) {
-		ms.automaticSnapshots = append(ms.automaticSnapshots, automaticSnapshotCall{InstanceName: si.InstanceName(), SnapConfig: cfg, Usernames: usernames, Flags: flags})
+	s.automaticSnapshots = nil
+	r := snapshotstate.MockBackendSave(func(_ context.Context, id uint64, si *snap.Info, cfg map[string]interface{}, usernames []string, flags *snapshotbackend.Flags) (*client.Snapshot, error) {
+		s.automaticSnapshots = append(s.automaticSnapshots, automaticSnapshotCall{InstanceName: si.InstanceName(), SnapConfig: cfg, Usernames: usernames, Flags: flags})
 		return nil, nil
 	})
+	s.AddCleanup(r)
 
-	restoreConnectRetryTimeout := ifacestate.MockConnectRetryTimeout(connectRetryTimeout)
-
-	ms.restore = func() {
-		snapstate.SetupRemoveHook = oldSetupRemoveHook
-		snapstate.SetupInstallHook = oldSetupInstallHook
-		restoreBackendSave()
-		restoreConnectRetryTimeout()
-	}
+	s.AddCleanup(ifacestate.MockConnectRetryTimeout(connectRetryTimeout))
 
 	os.Setenv("SNAPPY_SQUASHFS_UNPACK_FOR_TESTS", "1")
```

Comment:
(nitpick) we *could* add the unset as a AddCleanup here and then we don't need an explict TearDown for this suite anymore afaict.